.cbmc-batch/stubs/aws_hash_iter_overrides.c

```diff
@@ -9,15 +9,34 @@
 #include <stdio.h>
 #include <stdlib.h>
 
+/* If the consumer of the iterator doesn't use the .element in the iterator, we can just leave it undef
+ * This is sound, as it gives you a totally nondet value every time you call the iterator, and free under CBMC.
+ * But if it is used, we need a way for the harness to specify valid values for the element, for example
+ * if they are copying values out of the table.
+ * They can do this by defining -DHASH_ITER_ELEMENT_GENERATOR=the_generator_fn
```

Comment:
I don't understand the type of the generator, and I don't understand what parts of the rval passed to it the generator can expect to have value what what part it is expected to set.

Reply:
Updated comment